plugins/modules/mongodb_replicaset.py

```diff
@@ -554,13 +554,11 @@ def main():
     debug = module.params['debug']
     cluster_cmd = module.params['cluster_cmd']
 
-    voting_members = []
-    for member in members:
-        if ("votes" in member and member["votes"]) or "votes" not in member:
-            voting_members.append(member)
+    # Count voting members
+    voting_members = sum([1 if m.get("votes", 1) == 1 else 0 for m in members])
```

Comment:
OK I guess I misremembered here...

"The number of votes a server will cast in a [replica set election](https://www.mongodb.com/docs/manual/core/replica-set-elections/#std-label-replica-set-elections). The number of votes each member has is either 1 or 0"

https://www.mongodb.com/docs/manual/reference/replica-configuration/

I thought a member could have >1 votes, perhaps in the past but not anymore. Still, the solution is probably more python.

Thanks for the fix!

Reply:
Small issue..

>  TASK [mongodb_balancer : Create replicaset1 with module] ***********************
> 
> fatal: [testhost]: FAILED! => {
>     "changed": false,
>     "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 141, in <module>\n  File \"<stdin>\", line 133, in _ansiballz_main\n  File \"<stdin>\", line 81, in invoke_module\n  File \"/usr/lib/python3.10/runpy.py\", line 224, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.10/runpy.py\", line 96, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.10/runpy.py\", line 86, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_community.mongodb.mongodb_replicaset_payload_awkb0psa/ansible_community.mongodb.mongodb_replicaset_payload.zip/ansible_collections/community/mongodb/plugins/modules/mongodb_replicaset.py\", line 614, in <module>\n  File \"/tmp/ansible_community.mongodb.mongodb_replicaset_payload_awkb0psa/ansible_community.mongodb.mongodb_replicaset_payload.zip/ansible_collections/community/mongodb/plugins/modules/mongodb_replicaset.py\", line 558, in main\n  File \"/tmp/ansible_community.mongodb.mongodb_replicaset_payload_awkb0psa/ansible_community.mongodb.mongodb_replicaset_payload.zip/ansible_collections/community/mongodb/plugins/modules/mongodb_replicaset.py\", line 558, in <listcomp>\nAttributeError: 'str' object has no attribute 'get'\n",
>     "module_stdout": "",
>     "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
>     "rc": 1

https://github.com/ansible-collections/community.mongodb/blob/9124598d2ae5b2798d808cd3cfea49f1f79ca0ed/tests/integration/targets/mongodb_balancer/tasks/main.yml#L45

The members parameter is sometimes a list of dicts and sometimes a list of strings.

Reply:
I've adapted the condition to return 1 if a member is not a dict.